src/topology/separation.lean

```diff
@@ -554,3 +554,79 @@ begin
+  -- in whatever component x lies in and hence will be a subset of either u or v.
+  suffices : ∃ (Z : set α), is_clopen Z ∧ x ∈ Z ∧ Z ⊆ u ∪ v,
+  { cases this with Z H,
+    rw [disjoint_iff_inter_eq_empty] at huv,
+    have H1 := is_clopen_inter_of_disjoint_cover_clopen H.1 H.2.2 hu hv huv,
+    rw [union_comm] at H,
+    rw [inter_comm] at huv,
+    have H2 := is_clopen_inter_of_disjoint_cover_clopen H.1 H.2.2 hv hu huv,
+    by_cases (x ∈ u),
+    -- The x ∈ u case.
+    { left,
+      suffices : (⋂ (Z : {Z : set α // is_clopen Z ∧ x ∈ Z}), ↑Z) ⊆ u,
+      { rw [inter_comm, ←set.disjoint_iff_inter_eq_empty] at huv,
+        replace hab : (⋂ (Z : {Z // is_clopen Z ∧ x ∈ Z}), ↑Z) ≤ a ∪ b := hab,
+        replace this : (⋂ (Z : {Z // is_clopen Z ∧ x ∈ Z}), ↑Z) ≤ u := this,
+        exact disjoint.left_le_of_le_sup_right hab (huv.mono this hbv) },
+      { apply subset.trans _ (inter_subset_right Z u),
+        apply Inter_subset (λ Z : {Z : set α // is_clopen Z ∧ x ∈ Z}, ↑Z)
+        ⟨Z ∩ u, by {split, exact H1, apply mem_inter H.2.1 h}⟩ } },
+    -- If x ∉ u, we get x ∈ v since x ∈ u ∪ v. The rest is then like the x ∈ u case.
+    have h1 : x ∈ v,
+    { cases (mem_union x u v).1 (mem_of_subset_of_mem (subset.trans hab
+        (union_subset_union hau hbv)) (mem_Inter.2 (λ i, i.2.2))) with h1 h1,
+      { exfalso, apply h, exact h1},
+      { exact h1} },
+    right,
+    suffices : (⋂ (Z : {Z : set α // is_clopen Z ∧ x ∈ Z}), ↑Z) ⊆ v,
+    { rw [←set.disjoint_iff_inter_eq_empty] at huv,
+      replace hab : (⋂ (Z : {Z // is_clopen Z ∧ x ∈ Z}), ↑Z) ≤ a ∪ b := hab,
+      replace this : (⋂ (Z : {Z // is_clopen Z ∧ x ∈ Z}), ↑Z) ≤ v := this,
+      exact disjoint.left_le_of_le_sup_left hab (huv.mono this hau) },
+    { apply subset.trans _ (inter_subset_right Z v),
+      apply Inter_subset (λ Z : {Z : set α // is_clopen Z ∧ x ∈ Z}, ↑Z)
+      ⟨Z ∩ v, by {split, exact H2, apply mem_inter H.2.1 h1}⟩ } },
+  -- Now we find the required Z. We utilize the fact that X \ u ∪ v will be compact,
+  -- so there must be some finite intersection of clopen neighbourhoods of X disjoint to it,
+  -- but a finite intersection of clopen sets is clopen so we let this be our Z.
+  have H1 := (is_compact.inter_Inter_nonempty (is_closed.compact
+    (is_closed_compl_iff.2 (is_open_union hu hv)))
+    (λ Z : {Z : set α // is_clopen Z ∧ x ∈ Z}, Z) (λ Z, Z.2.1.2)),
+  rw [←not_imp_not, not_forall, not_nonempty_iff_eq_empty, inter_comm] at H1,
+  have huv_union := subset.trans hab (union_subset_union hau hbv),
+  rw [←set.compl_compl (u ∪ v), subset_compl_iff_disjoint] at huv_union,
+  replace H1 := H1 huv_union,
+  cases H1 with Zi H2,
+  existsi (⋂ (U ∈ Zi), subtype.val U),
+  split,
+    { apply @is_clopen_bInter _ _ _ _ _ _, exact (λ Z hZ, Z.2.1) },
+    { split,
+      { exact mem_bInter_iff.2 (λ Z hZ, Z.2.2) },
+      { rw [not_nonempty_iff_eq_empty, inter_comm, ←subset_compl_iff_disjoint] at H2,
+        rw [compl_compl] at H2, exact H2 } },
```

Comment:
One last example of how such a proof can be streamlined:
- you have a replace, and then you case on the new `H1`. You can directly case on `H1 huv_union`.
- you have a split, and then a second split. Instead, you could do the two splits directly, with a refine instead of existsi.
- The indentation after the split is strange.
- The `@is_clopen_bInter` is not necessary, `is_clopen_bInter` works fine.

After cleaning all this, it becomes
```suggestion
  cases H1 huv_union with Zi H2,
  refine ⟨(⋂ (U ∈ Zi), subtype.val U), _, _, _⟩,
  { exact is_clopen_bInter (λ Z hZ, Z.2.1) },
  { exact mem_bInter_iff.2 (λ Z hZ, Z.2.2) },
  { rwa [not_nonempty_iff_eq_empty, inter_comm, ←subset_compl_iff_disjoint, compl_compl] at H2 }
```

Reply:
Thank you! It seems I need to utilize refine more :)

Reply:
That's the point of detailed PR reviews like that: teach you a lot of tricks that you may use in your next PRs!